piptools/repositories/pypi.py

```diff
@@ -226,9 +226,16 @@ def resolve_reqs(self, download_dir, ireq, wheel_cache):
                 reqset = RequirementSet()
                 ireq.is_direct = True
                 reqset.add_requirement(ireq)
+
                 resolver = PipResolver(**resolver_kwargs)
-                resolver.require_hashes = False
-                results = resolver._resolve_one(reqset, ireq)
+                require_hashes = False
+                if PIP_VERSION < (19, 4):
+                    resolver.require_hashes = require_hashes
+                    results = resolver._resolve_one(reqset, ireq)
+                else:  # pragma: no cover
+                    # TODO remove pragma after pip==19.4 being released
+                    results = resolver._resolve_one(reqset, ireq, require_hashes)
```

Comment:
Would be better to not have "pragma: no cover" there in the first place, especially since this should be covered already via the pipmaster job?!
(but even if not: it is better to clearly indicate that some code is not covered)

Reply:
Thanks for getting attention to this!

What the options do we have to avoid arbitrarily decreasing/increasing the test coverage like in this https://github.com/jazzband/pip-tools/pull/909#issuecomment-535671204 or #914 PRs? It would be annoying for users (maintainers too) to see that PR with 100% coverage has failed checks because of the other code, which is genuinely covered but temporarily is not, because of the pip master job fails at import time.

On the other hand, we do have a relative coverage check (codecov/patch), so we can rely entirely on it and don't pay attention to the absolute (codecov/project) test coverage.

Reply:
@blueyed would you like to open an issue for that?

Reply:
@atugushev 
No it's fine for now I guess - I'd rather ignore any "X" from CI and override it after checking (about coverage,e.g.via codecov).

Reply:
Okay, I'll remove the TODO in the following pip-master compatibility PR.

Reply:
FTR, addressed in #978 (fa2c479dd19f73fd544abbaaef624357d41b6f15)